### ISSUE TITLE
Users.changeset isn't usable for updates

### ISSUE THREAD

**Issue by dfalling** [AppDoctorIo/accesspass#33]
Currently `Users.changeset` only casts parameters, leaving all validation to the `create_user_changeset` function which also does other tasks like creating a user's ID.  Is it possible we could move the validation to the changeset function so it can be leveraged for updates as well?

Something like this:
```elixir
  def changeset(schema, params \\ :empty) do
    schema
    |> cast(params, @required_fields ++ @optional_fields)
    |> validate_email
    |> validate_required([:username, :email, :password, :password_confirm])
    |> validate_length(:username, min: 3, max: 36)
    |> validate_length(:email, min: 3, max: 256)
    |> validate_length(:password, min: 6)
    |> compare_passwords()
    |> unique_constraint(:email)
    |> unique_constraint(:user_id)
    |> unique_constraint(:username)
  end
```

**Comment by jpiepkow** [comment]
I think this is a reasonable change. I will look to try and add it tonight. Thanks!

**Comment by dfalling** [comment]
Great, thanks!

**Comment by jpiepkow** [comment]
resolved in 0.6.3 Let me know if that works for you.

**Comment by dfalling** [comment]
It's really close.  The one issue I'm hitting is `password` and `password_confirm` are required in the changeset forcing you to resubmit them for updates.

I'll look more into this tomorrow.  I think with Ecto it may work if `create_user_changeset` requires password, and `changeset` has something equivalent to `Ecto.Changeset.validate_confirmation(:password)`.  I'll investigate it and PR.
